pkg/agent/controller/egress/egress_controller.go

```diff
@@ -521,14 +522,34 @@ func (c *EgressController) unbindPodEgress(pod, egress string) (string, bool) {
 	return "", false
 }
 
-func (c *EgressController) updateEgressStatus(egress *crdv1a2.Egress, nodeName string) error {
-	if egress.Status.EgressNode == nodeName {
-		return nil
-	}
-	klog.V(2).InfoS("Updating Egress status", "Egress", egress.Name, "oldNode", egress.Status.EgressNode, "newNode", nodeName)
+func (c *EgressController) updateEgressStatus(egress *crdv1a2.Egress, isLocal bool) error {
 	toUpdate := egress.DeepCopy()
-	toUpdate.Status.EgressNode = nodeName
-	if _, err := c.crdClient.CrdV1alpha2().Egresses().UpdateStatus(context.TODO(), toUpdate, metav1.UpdateOptions{}); err != nil {
+	nodeNameToUpdate := c.nodeName
+	if !isLocal {
+		if egress.Status.EgressNode != c.nodeName {
+			return nil
+		}
+		nodeNameToUpdate = ""
+	}
```

Comment:
this should be moved to the loop below and handle `isLocal=true` case as well, it doesn't make to separate the two cases and add a special check for `isLocal` case outside of the loop alone.